core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -96,6 +98,28 @@ public static MetricsConfig forTable(Table table) {
     return from(table.properties(), table.sortOrder());
   }
 
+  /**
+   * Creates a metrics config for a position delete file.
+   *
+   * @param table an Iceberg table
+   */
+  public static MetricsConfig forPositionDelete(Table table) {
+    MetricsConfig config = new MetricsConfig();
+
+    config.columnModes.put(MetadataColumns.DELETE_FILE_PATH.name(), MetricsModes.Full.get());
+    config.columnModes.put(MetadataColumns.DELETE_FILE_POS.name(), MetricsModes.Full.get());
+
+    MetricsConfig tableConfig = forTable(table);
+
+    config.defaultMode = tableConfig.defaultMode;
+    tableConfig.columnModes.forEach((columnAlias, mode) -> {
+      String newColumnAlias = DOT.join(MetadataColumns.DELETE_FILE_ROW_FIELD_NAME, columnAlias);
```

Comment:
Maybe this is too much future planning, but since we'll need to look up delete file column aliases via this same transformation in the future, maybe we should codify it?

public static String deleteMetricAlias(String columnName) {
  return Dot.join(MetadataColumns.DELETE_FILE_ROW_FIELD_NAME, columnName)
}

Reply:
I can definitely do that. We will have to call it `positionDeleteXXX` as the shift is only position delete files.

Reply:
After thinking more, I've decided not to expose it just yet. We have been debating whether to switch to column IDs and since there is no clarity on what changes will be required and this particular functionality is not needed anywhere outside of this class, it seems safer to keep it internal. We can always expose once needed.

I did change the var name, though.

Reply:
Sgtm